src/schemas/json/sublime-syntax.json

```diff
@@ -0,0 +1,249 @@
+          "type": "boolean"
+        }
+      }
+    },
+    "match-pattern": {
+      "description": "The following keys control behavior that is exclusive, and only one can be specified per match pattern: push pop set embed branch fail\nhttps://www.sublimetext.com/docs/syntax.html#match-patterns",
+      "type": "object",
+      "properties": {
+        "match": {
+          "description": "The regex used to match against the text. YAML allows many strings to be written without quotes, which can help make the regex clearer, but it’s important to understand when you need to quote the regex. If your regex includes the characters #, :, -, {, [ or > then you likely need to quote it. Regexes are only ever run against a single line of text at a time.",
+          "type": "string"
+        },
+        "scope": {
+          "description": "The scope assigned to the matched text.",
+          "pattern": "^[_a-zA-Z][a-zA-Z0-9._-]*$",
+          "type": "string"
+        },
+        "captures": {
+          "description": "A mapping of numbers to scope, assigning scopes to captured portions of the match regex.",
+          "type": "object",
+          "properties": {
+            "patternProperties": {
+              "^\\d+$": {
+                "type": "string"
+              }
+            }
+          },
+          "minProperties": 1
+        },
+        "push": {
+          "description": "The contexts to push onto the stack. This may be either a single context name, a list of context names, or an inline, anonymous context.",
+          "anyOf": [
+            {
+              "type": "string"
+            },
+            {
+              "type": "array",
+              "items": {
+                "type": "string"
+              },
+              "minItems": 1,
+              "uniqueItems": true
+            },
+            {
+              "$ref": "#/definitions/patterns"
+            }
+          ]
+        },
+        "pop": {
+          "description": "Pops contexts off the stack. The value true will pop a single context. An integer greater than zero will pop the corresponding number of contexts. The pop key can be combined with push, set, embed and branch. When combined, the specified number of contexts will be popped off of the stack before the other action is performed. For push, embed and branch actions, the pop treats the match as if it were a lookahead, which means the match will not receive the meta_scope of the contexts that are popped.",
+          "type": "boolean"
+        },
+        "set": {
+          "description": "Accepts the same arguments as push, but will first pop this context off, and then push the given context(s) onto the stack. Any match will receive the meta_scope of the context being popped and the context being pushed.",
+          "type": "string"
```

Comment:
This should be the same as for `push` - allow an array of strings or an object representing an anonymous context etc